From: Constantin Piber <cp.piber@gmail.com>
Date: Tue, 18 Nov 2025 15:43:15 +0800
Subject: Avoid #embed directive

Workaround gcc-14's lack of #embed support. This can be dropped when Debian's
default gcc is 15+.

Origin: https://github.com/cpiber/hyprland-ppa/blob/0dcd5f3101f7ced40879d0f34f337b1a2cdd79ae/hyprland/debian/patches/03-embed
Forwarded: not-needed
---
 CMakeLists.txt                   | 5 +++++
 src/config/lua/DefaultConfig.hpp | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9ef49eb..3c525f6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,6 +33,11 @@ if(NOT HYPR_SHADER_GEN_RESULT EQUAL 0)
   )
 endif()
 
+execute_process(COMMAND xxd -i
+                INPUT_FILE ./example/hyprland.lua
+                OUTPUT_FILE ./src/config/lua/DefaultConfigBytes.hpp
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+
 find_package(PkgConfig REQUIRED)
 find_package(Python3 COMPONENTS Interpreter QUIET)
 
diff --git a/src/config/lua/DefaultConfig.hpp b/src/config/lua/DefaultConfig.hpp
index 8911a8c..4db65b9 100644
--- a/src/config/lua/DefaultConfig.hpp
+++ b/src/config/lua/DefaultConfig.hpp
@@ -12,7 +12,7 @@ hl.config({ autogenerated = true }) -- remove this line to remove the warning
 
 )#";
 inline constexpr char             EXAMPLE_CONFIG_BYTES_LUA[] = {
-#embed "../../../example/hyprland.lua"
+#include "DefaultConfigBytes.hpp"
 };
 
 inline constexpr std::string_view EXAMPLE_CONFIG_LUA = {EXAMPLE_CONFIG_BYTES_LUA, sizeof(EXAMPLE_CONFIG_BYTES_LUA)};
